Description: Don't install files removed from repacked upstream tarball.
Author: Bas Couwenberg <sebastic@debian.org>
Forwarded: not-needed

--- a/gdal.cmake
+++ b/gdal.cmake
@@ -660,9 +660,7 @@ set(GDAL_DATA_FILES
     data/GDALLogoColor.svg
     data/GDALLogoGS.svg
     data/bag_template.xml
-    data/cubewerx_extra.wkt
     data/default.rsc
-    data/ecw_cs.wkt
     data/eedaconf.json
     data/epsg.wkt
     data/esri_StatePlane_extra.wkt
